src/__tests__/App.test.jsx

```diff
@@ -70,28 +103,89 @@ describe("App component", () => {
     expect(dayInputBox.value).toEqual('10')
   });
 
-  it("prompts the user to input the month", () => {
+  it("doesn't allow the user to input day less than 1", () => {
     render(<App />);
-    expect(screen.getByText("Month:")).toBeDefined()
+    const dayInputBox = screen.getByLabelText("day-input");
+
+    fireEvent.change(dayInputBox, {target: {value: '0'}})
+
+    expect(dayInputBox.value).toEqual('1')
   });
 
-  it("has an input box for the month", () => {
+  it("doesn't allow the user to input day greater than 31", () => {
     render(<App />);
-    expect(screen.getByLabelText("month-input")).toBeDefined()
+    const dayInputBox = screen.getByLabelText("day-input");
+
+    fireEvent.change(dayInputBox, {target: {value: '40'}})
+
+    expect(dayInputBox.value).toEqual('31')
   });
 
-  it("allows user to update bday month", () => {
+
+  it("has a button for user to submit info when completed", () => {
     render(<App />);
-    const monthInputField = screen.getByLabelText("month-input")
+    expect(screen.getByRole("button")).toBeDefined()
+  });
+
+  it("tells the user happy birthday if today is their birthday", () => {
+    const todayDate = new Date('2024-12-31')
+    vi.setSystemTime(todayDate)
+    render(<App />);
+    fireEvent.change(screen.getByPlaceholderText("name"), {target: {value: 'Test Name'}})
+    fireEvent.change(screen.getByLabelText("month-input"), {target: {value: '2024-12'}})
+    fireEvent.change(screen.getByLabelText("day-input"), {target: {value: '31'}})
+    fireEvent.click(screen.getByRole("button"))
+
+    expect(screen.getByText("Happy Birthday Test Name!")).toBeDefined();
+
+  });
+
+  it("doesn't have a happy birthday message before submission", () => {
+    render(<App />);
+    fireEvent.change(screen.getByPlaceholderText("name"), {target: {value: 'Test Name'}})
+    fireEvent.change(screen.getByLabelText("month-input"), {target: {value: '2024-01'}})
+    fireEvent.change(screen.getByLabelText("day-input"), {target: {value: '01'}})
     
-    fireEvent.change(monthInputField, {target: {value: "2020-01"}})
+    expect(screen.queryByText("Happy Birthday Test Name!")).toBeNull();
 
-    expect(monthInputField.value).toBe("2020-01")
   });
 
-  it("has a button for user to submit info when completed", () => {
+  it("tells the user how many days until next birthday", () => {
     render(<App />);
-    expect(screen.getByRole("button")).toBeDefined()
-  })
+    fireEvent.change(screen.getByPlaceholderText("name"), {target: {value: 'Test Name'}})
+    fireEvent.change(screen.getByLabelText("month-input"), {target: {value: '2024-01'}})
+    fireEvent.change(screen.getByLabelText("day-input"), {target: {value: '07'}})
+    fireEvent.click(screen.getByRole("button"))
```

Comment:
Could you write a helper function that does this processing and you just pass in the parameters that change?